src/migratus/cli.clj

```diff
@@ -0,0 +1,150 @@
+(ns migratus.cli
+  (:require [clojure.java.io :as io]
+            [clojure.string :as str]
+            [clojure.tools.cli :refer [parse-opts]]
+            [clojure.tools.logging :as log]
+            [migratus.core :as migratus])
+  (:import [java.time ZoneOffset]
+           [java.util.logging ConsoleHandler Logger SimpleFormatter]))
+
+(def global-cli-options
+  [[nil "--config NAME" "Configuration file name" :default "migratus.edn"]
+   ["-h" "--help"]])
+
+(def migrate-cli-options
+  [[nil "--until-just-before MIGRATION-ID" "Run all migrations preceding migration-id. This is useful when testing that a migration behaves as expected on fixture data. This only considers uncompleted migrations, and will not migrate down."]
+   ["-h" "--help"]])
+
+(def rollback-cli-options
+  [[nil "--until-just-after MIGRATION-ID" "Migrate down all migrations after migration-id. This only considers completed migrations, and will not migrate up."]
+   ["-h" "--help"]])
+
+(def list-cli-options
+  [[nil "--available" "List all migrations, applyed and non applyed"]
+   [nil "--pending" "List pending migrations"]
+   [nil "--applyed" "List applyed migrations"]
+   ["-h" "--help"]])
+
+(defn usage [options-summary]
+  (->> ["Usage: migratus action [options]"
+        ""
+        "Actions:"
+        "  init"
+        "  create"
+        "  migrate"
+        "  reset"
+        "  rollback"
+        "  up"
+        "  down"
+        "  list"
+        ""
+        "options:"
+        options-summary]
+       (str/join \newline)))
+
+(defn error-msg [errors]
+  (log/info "The following errors occurred while parsing your command:\n\n"
+            (str/join  \newline errors)))
+
+(defn no-match-message
+  "No matching clause message info"
+  [arguments summary]
+  (log/info "Migratus API does not support this action(s) : " arguments "\n\n"
+            (str/join (usage summary))))
+
+(defn run-migrate [cfg args]
```

Comment:
minor nitpick, we could just destructure as `[_ & args]` here instead of making `rargs`